src/models/node.ts

```diff
@@ -363,6 +423,139 @@ export class Node<N extends INodeBase, E extends IEdgeBase> implements INode<N,
     });
   }
 
+  addListener(observer: IObserver): void {
+    this._listeners.push(observer);
+  }
+
+  removeListener(observer: IObserver): void {
+    this._listeners.splice(this._listeners.indexOf(observer), 1);
+  }
+
+  notifyListeners(): void {
+    this._listeners.forEach((listener) => {
+      listener.update();
+    });
+  }
+
+  setData(data: N): void;
+
+  setData(callback: (node: INode<N, E>) => N): void;
+
+  setData(arg: N | ((node: INode<N, E>) => N)) {
+    if (typeof arg === 'function') {
+      this._data = (arg as (node: INode<N, E>) => N)(this);
+    } else {
+      this._data = arg as N;
+    }
+    this.notifyListeners();
+  }
+
+  patchData(data: Partial<N>): void;
+
+  patchData(callback: (node: INode<N, E>) => Partial<N>): void;
+
+  patchData(arg: Partial<N> | ((node: INode<N, E>) => Partial<N>)) {
+    let data: Partial<N>;
+
+    if (typeof arg === 'function') {
+      data = (arg as (node: INode<N, E>) => Partial<N>)(this);
+    } else {
+      data = arg as Partial<N>;
+    }
+
+    Object.keys(data).forEach((key) => {
+      // @ts-ignore
+      this._data[key] = data[key];
+    });
+
+    this.notifyListeners();
+  }
+
+  setPosition(position: INodeCoordinates | INodeMapCoordinates | INodePosition): void;
+
+  setPosition(callback: (node: INode<N, E>) => INodeCoordinates | INodeMapCoordinates | INodePosition): void;
+
+  setPosition(
+    arg:
+      | INodeCoordinates
+      | INodeMapCoordinates
+      | INodePosition
+      | ((node: INode<N, E>) => INodeCoordinates | INodeMapCoordinates | INodePosition),
```

Comment:
Should I merge these interfaces somehow? I was considering assigning a type to nodes (for map view and regular view), but I'm unsure if it's necessary since users can use any value from the node data for `lng` and `lat`. Is there a better approach to setting the position for map view nodes maybe?